relays/ethereum/src/ethereum_sync_loop.rs

```diff
@@ -30,41 +30,30 @@ const SUBSTRATE_TICK_INTERVAL_MS: u64 = 5_000;
 
 /// Ethereum synchronization parameters.
 pub struct EthereumSyncParams {
-	/// Ethereum RPC host.
-	pub eth_host: String,
-	/// Ethereum RPC port.
-	pub eth_port: u16,
-	/// Substrate RPC host.
-	pub sub_host: String,
-	/// Substrate RPC port.
-	pub sub_port: u16,
-	/// Substrate transactions signer.
-	pub sub_signer: sp_core::sr25519::Pair,
+	/// Ethereum connection params.
+	pub eth: EthereumConnectionParams,
+	/// Substrate connection params.
+	pub sub: SubstrateConnectionParams,
+	/// Substrate signing params.
+	pub sub_sign: SubstrateSigningParams,
 	/// Synchronization parameters.
 	pub sync_params: HeadersSyncParams,
 }
 
-impl std::fmt::Debug for EthereumSyncParams {
-	fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-		f.debug_struct("EthereumSyncParams")
-			.field("eth_host", &self.eth_host)
-			.field("eth_port", &self.eth_port)
-			.field("sub_host", &self.sub_port)
-			.field("sub_port", &self.sub_port)
-			.field("sync_params", &self.sync_params)
-			.finish()
-	}
-}
-
 impl Default for EthereumSyncParams {
 	fn default() -> Self {
 		EthereumSyncParams {
-			eth_host: "localhost".into(),
-			eth_port: 8545,
-			sub_host: "localhost".into(),
-			sub_port: 9933,
-			sub_signer: sp_keyring::AccountKeyring::Alice.pair(),
-			sync_params: Default::default(),
+			eth: Default::default(),
+			sub: Default::default(),
+			sub_sign: Default::default(),
+			sync_params: HeadersSyncParams {
+				max_future_headers_to_download: 128,
+				max_headers_in_submitted_status: 128,
+				max_headers_in_single_submit: 32,
+				max_headers_size_in_single_submit: 131_072,
```

Comment:
We should make this a `const` and have the explanation you gave me for how to get to this number as a comment